python/example_code/s3/scenarios/conditional_requests/scenario.py

```diff
@@ -0,0 +1,299 @@
+        """
+        Display a list of the objects in the test buckets.
+
+        :param source_bucket: The name of the source bucket.
+        :param dest_bucket: The name of the destination bucket.
+        """
+        self.list_bucket_contents(source_bucket)
+        self.list_bucket_contents(dest_bucket)
+
+    def list_bucket_contents(self, bucket_name):
+        """
+        Display a list of the objects in the bucket.
+
+        :param bucket_name: The name of the bucket.
+        """
+        try:
+            # Get list of all objects in the bucket.
+            print(f"\t Items in bucket {bucket_name}")
+            list_response = self.s3_client.list_objects_v2(Bucket=bucket_name)
+            objs = list_response.get("Contents", [])
+            if not objs:
+                print("\t\tNo objects found.")
+            for obj in objs:
+                key = obj["Key"]
+                print(f"\t\t object: {key} ETag {obj['ETag']}")
+            return objs
+        except ClientError as e:
+            error_code = e.response["Error"]["Code"]
+            if error_code == "NoSuchBucket":
+                logger.info(f"Bucket {bucket_name} does not exist.")
+            else:
+                logger.error(f"Error listing bucket and objects: {error_code}")
+                raise
+
+    # snippet-start:[python.example_code.s3.DisplayMenu]
+
+    def display_menu(
+        self, source_bucket: str, dest_bucket: str, object_key: str, etag: str
+    ):
+        """
+        Displays the menu of conditional request options for the user.
+
+        :param source_bucket: The name of the source bucket.
+        :param dest_bucket: The name of the destination bucket.
+        :param object_key: The key of the test object in the source bucket.
+        :param etag: The etag of the test object in the source bucket.
+        """
+
+        actions = [
+            "Print list of bucket items.",
+            "Perform a conditional read.",
+            "Perform a conditional copy.",
+            "Perform a conditional write.",
+            "Clean up and exit.",
+        ]
+
+        conditions = [
+            "If-Match: using the object's ETag. This condition should succeed.",
+            "If-None-Match: using the object's ETag. This condition should fail.",
+            "If-Modified-Since: using yesterday's date. This condition should succeed.",
+            "If-Unmodified-Since: using yesterday's date. This condition should fail.",
+        ]
+
+        condition_types = [
+            "IfMatch",
+            "IfNoneMatch",
+            "IfModifiedSince",
+            "IfUnmodifiedSince",
+        ]
+        copy_condition_types = [
+            "CopySourceIfMatch",
+            "CopySourceIfNoneMatch",
+            "CopySourceIfModifiedSince",
+            "CopySourceIfUnmodifiedSince",
+        ]
+
+        yesterday_date = datetime.datetime.utcnow() - datetime.timedelta(days=1)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Naive datetime objects are treated by many datetime methods as local times, We recommend that you create aware timestamps with the `now` or `fromtimestamp` functions using the `tzinfo` to attribute to represent times in UTC. Do not use the [datetime.utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) because it returns only a `naive` datetime object. 

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)